examples/cookbook/utils/check-account-existence.js

```diff
@@ -0,0 +1,28 @@
+// Demonstrates checking if an account exists
+const { providers } = require("near-api-js");
+
+const provider = new providers.JsonRpcProvider(
+    "https://archival-rpc.testnet.near.org"
+);
+
+async function accountExists() {
+    let rawResult;
+
+    for (const account_id of ["does-not-exist.mike.testnet", "mike.testnet"]) {
+        let succeeded = true;
+        try {
+            rawResult = await provider.query({
+                request_type: "view_account",
+                account_id: account_id,
+                finality: "final",
+            });
+        } catch (e) {
+            if (e.type === 'AccountDoesNotExist') {
+                succeeded = false;
+            }
+        }
```

Comment:
I am not sure if `view_account` can throw anything else, but shouldn't we rethrow an exception is its of an unexpected type? Otherwise we might report false-positive results in this function.